core/src/main/java/com/datastax/oss/driver/api/core/config/ProgrammaticDriverConfigLoader.java

```diff
@@ -0,0 +1,217 @@
+      return with(profileName, option, value);
+    }
+
+    public Builder withString(DriverOption option, String value) {
+      return with(option, value);
+    }
+
+    public Builder withString(String profileName, DriverOption option, String value) {
+      return with(profileName, option, value);
+    }
+
+    public Builder withStringList(DriverOption option, List<String> value) {
+      return with(option, value);
+    }
+
+    public Builder withStringList(String profileName, DriverOption option, List<String> value) {
+      return with(option, value);
+    }
+
+    public Builder withStringMap(DriverOption option, Map<String, String> value) {
+      for (String key : value.keySet()) {
+        config =
+            config.withValue(
+                option.getPath() + '.' + key, ConfigValueFactory.fromAnyRef(value.get(key)));
+      }
+      return this;
+    }
+
+    public Builder withStringMap(
+        String profileName, DriverOption option, Map<String, String> value) {
+      for (String key : value.keySet()) {
+        String path = "profiles." + profileName + '.' + option.getPath() + '.' + key;
+        config = config.withValue(path, ConfigValueFactory.fromAnyRef(value.get(key)));
+      }
+      return this;
+    }
+
+    public Builder withBytes(DriverOption option, long value) {
+      return with(option, value);
+    }
+
+    public Builder withBytes(String profileName, DriverOption option, long value) {
+      return with(profileName, option, value);
+    }
+
+    public Builder withBytesList(DriverOption option, List<Long> value) {
+      return with(option, value);
+    }
+
+    public Builder withBytesList(String profileName, DriverOption option, List<Long> value) {
+      return with(profileName, option, value);
+    }
+
+    public Builder withDuration(DriverOption option, Duration value) {
+      return with(option, value);
+    }
+
+    public Builder withDuration(String profileName, DriverOption option, Duration value) {
+      return with(profileName, option, value);
+    }
+
+    public Builder withDurationList(DriverOption option, List<Duration> value) {
+      return with(option, value);
+    }
+
+    public Builder withDurationList(String profileName, DriverOption option, List<Duration> value) {
+      return with(profileName, option, value);
+    }
+
+    public Builder withClass(DriverOption option, Class<?> value) {
+      return with(option, value.getName());
+    }
+
+    public Builder withClass(String profileName, DriverOption option, Class<?> value) {
+      return with(profileName, option, value.getName());
+    }
+
+    /** Unsets an option. */
+    public Builder without(DriverOption option) {
+      return with(option, null);
+    }
+
+    public Builder without(String profileName, DriverOption option) {
+      return with(profileName, option, null);
+    }
+
+    public Builder withFallback(Supplier<Config> configSupplier) {
```

Comment:
Hrm, i'll think some more about this.  Agree that we shouldn't leak typesafe config.  In addition I still need to think a little bit more about how this will work with the dse driver.  I think preferably this could be part of api and not internal, but maybe that is the right thing to do if there isn't a good way to allow overriding the backing config in some way.